analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarSyntaxNodeReportingContext.cs

```diff
@@ -49,6 +49,16 @@ public sealed class SonarSyntaxNodeReportingContext : SonarTreeReportingContextB
     public bool IsAzureFunction() =>
         AzureFunctionMethod() is not null;
 
+    public override bool ShouldAnalyzeTree(SyntaxTree tree, GeneratedCodeRecognizer generatedCodeRecognizer)
+    {
+        if (GeneratedCodeRecognizer.IsRazorGeneratedFile(tree) && !Node.GetLocation().GetMappedLineSpan().HasMappedPath)
+        {
+            return false;
+        }
+
+        return base.ShouldAnalyzeTree(tree, generatedCodeRecognizer);
```

Comment:
tree.IsConsideredGenerated in the base class will take care of cases when razor files should or should not be generated. Also in future PRs, the base class will take care of checking the scanner configuration, see here [this](https://github.com/SonarSource/sonar-dotnet/pull/7917) PR for more context. My suggestion is to not change that behaviour and to restrict the changes in this PR to the `ReportIssueCore` method.

Reply:
This change should prevent the analysis from happening if the node cannot be mapped back to the original file even when the razor analysis is enabled.
Otherwise, the analysis will happen to be discarded in `ReportIssueCore`.

Since a majority of our rules rely on `SyntaxNode`, I believe it is a nice addition to be able to not analyze part of the code that won't be reported anyway.